app/react/src/client/preview/render.tsx

```diff
@@ -7,14 +7,16 @@ import { RenderMainArgs } from './types';
 
 const rootEl = document ? document.getElementById('root') : null;
 
-function render(node: React.ReactElement, el: Element) {
-  ReactDOM.render(
-    process.env.STORYBOOK_EXAMPLE_APP ? <React.StrictMode>{node}</React.StrictMode> : node,
-    el
-  );
-}
+const render = (node: React.ReactElement, el: Element) =>
+  new Promise(resolve => {
+    ReactDOM.render(
+      process.env.STORYBOOK_EXAMPLE_APP ? <React.StrictMode>{node}</React.StrictMode> : node,
+      el,
+      resolve
+    );
+  });
 
-export default function renderMain({
+export default async function renderMain({
```

Comment:
Do the other renderers also need to be updated, or just React?

Reply:
React app is specific because it doesn't call `storyFn`